src/Admin/components/Notices/index.tsx

```diff
@@ -0,0 +1,56 @@
+import {useState} from '@wordpress/element';
+import {__} from '@wordpress/i18n';
+import {WarningIcon, ErrorIcon, InfoIcon, CloseIcon} from '@givewp/admin/components/Notices/Icons';
+import styles from './styles.module.scss';
+
+/**
+ * @unreleased
+ */
+interface Props {
+    type: 'info' | 'warning' | 'error';
+    children: React.ReactNode;
+    dismissHandleClick?: () => void;
+}
+
+/**
+ * @unreleased
+ */
+export default ({type, children, dismissHandleClick}: Props) => {
+    const [isVisible, setIsVisible] = useState(true);
+    
+    const handleDismiss = () => {
+        setIsVisible(false);
+        if (dismissHandleClick) {
+            dismissHandleClick();
+        }
+    };
+    
+    if (!isVisible) {
+        return null;
+    }
+    
+    const noticeClasses = `${styles.notice} ${
+        type === 'warning' ? styles.warning : type === 'error' ? styles.error : styles.info
+    }`;
+    
+    return (
+        <div className={noticeClasses}>
+            {type === 'warning' && <WarningIcon />}
+            {type === 'error' && <ErrorIcon />}
+            {type === 'info' && <InfoIcon />}
```

Comment:
Nothing wrong here but another possible solution could be something like:

```js
const noticeClasses = `${styles.notice} ${styles[type] ?? styles.info}`;

const NoticeIcon = ({ type }: { type: Props['type'] }) => {
    const icons = {
        warning: WarningIcon,
        error: ErrorIcon,
        info: InfoIcon
    };
    const IconComponent = icons[type] ?? icons.info;
    return <IconComponent />;
};
```

then
```js
<NoticeIcon type={type} />
```